torchaudio/transforms/_transforms.py

```diff
@@ -2089,3 +2089,58 @@ def forward(
 
         specgram_enhanced.to(dtype)
         return specgram_enhanced
+
+
+class SoudenMVDR(torch.nn.Module):
+    r"""Minimum Variance Distortionless Response (*MVDR* [:footcite:`capon1969high`]) module
+    based on the method proposed by *Souden et, al.* [:footcite:`souden2009optimal`].
+
+    .. devices:: CPU CUDA
+
+    .. properties:: Autograd TorchScript
+
+    .. math::
+        \textbf{w}_{\text{MVDR}}(f) =
+        \frac{{{\bf{\Phi}_{\textbf{NN}}^{-1}}(f){\bf{\Phi}_{\textbf{SS}}}}(f)}
+        {\text{Trace}({{{\bf{\Phi}_{\textbf{NN}}^{-1}}(f) \bf{\Phi}_{\textbf{SS}}}(f))}}\bm{u}
+
+    where :math:`\bf{\Phi}_{\textbf{SS}}` and :math:`\bf{\Phi}_{\textbf{NN}}`
+    are the power spectral density (PSD) matrices of speech and noise, respectively.
+    :math:`\bf{u}` is a one-hot vector that represents the reference channel.
+    """
+
+    def forward(
+        self,
+        specgram: Tensor,
+        psd_s: Tensor,
+        psd_n: Tensor,
+        reference_channel: Union[int, Tensor],
+        diagonal_loading: bool = True,
+        diag_eps: float = 1e-7,
+        eps: float = 1e-8,
+    ) -> Tensor:
+        """
+        Args:
+            specgram (Tensor): Multi-channel complex-valued spectrum.
+                Tensor of dimension `(..., channel, freq, time)`
+            psd_s (Tensor): The complex-valued power spectral density (PSD) matrix of target speech.
+                Tensor of dimension `(..., freq, channel, channel)`
+            psd_n (Tensor): The complex-valued power spectral density (PSD) matrix of noise.
+                Tensor of dimension `(..., freq, channel, channel)`
+            reference_channel (int or Tensor): Indicate the reference channel.
+                If the dtype is ``int``, it represent the reference channel index.
+                If the dtype is ``Tensor``, the dimension is `(..., channel)`, where the ``channel`` dimension
+                is one-hot.
+            diagonal_loading (bool, optional): whether to apply diagonal loading to psd_n
+                (Default: ``True``)
+            diag_eps (float, optional): The coefficient multiplied to the identity matrix for diagonal loading
+                (Default: ``1e-7``)
+            eps (float, optional): a value added to the denominator in the beamforming weight computation.
```

Comment:
nit: nicer to stick to one standard for capitalization
```suggestion
            eps (float, optional): Value to add to the denominator in the beamforming weight formula.
```